src/types.rs

```diff
@@ -16,14 +16,35 @@ impl AccountId {
 
     // The digit portion of the account hint.
     pub fn hint(&self) -> &str {
-        &self.0[0..3]
+        // We don't currently enforce correctness for the account ID so this method needs to be forgiving.
+
+        match self.0.char_indices().nth(3) {
+            Some((i, _)) => &self.0[..i],
+            None => &self.0,
+        }
```

Comment:
same result, but maybe `self.0.get(0..3).unwrap_or(&self.0)` would be nicer

Reply:
I don't think that is the same result if `3` isn't a valid character index.

Reply:
Yes, the test fails:

```
---- types::test_account_id_hint stdout ----
thread 'types::test_account_id_hint' panicked at src/types.rs:42:5:
assertion `left == right` failed
  left: "😛🤠🙃😉"
 right: "😛🤠🙃"
```

Reply:
oh, I missed that you are using char indices... I thought you were just returning the whole thing if there's no char boundary after the 3rd byte.